packages/sdk/src/types/runtime/actor.ts

```diff
@@ -98,8 +99,11 @@ export class Actor implements ActorLike, Patchable<ActorLike> {
     private _text: Text;
     private _attachment: Attachment;
     private _lookAt: LookAt;
+    private _grabbable = false;
     // tslint:enable:variable-name
 
+    private grab = new DiscreteAction();

```

Comment:
>grab [](start = 12, length = 4)

Most objects won't be grabbable. Can `grab` be lazily allocated?

Reply:
I don't see why not, but we don't actually do that anywhere else in the behavior system at the moment.

Reply:
I don't see why not, but we don't actually do that anywhere else in the behavior system at the moment.

Reply:
Curious: Would `grab` become a component once we have that system in place?

---
In reply to: [266113938](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/260#discussion_r266113938) [](ancestors = 266113938)

Reply:
I don't know.  My initial gut reaction would be possibly, but it is more like a property of the actor.  Not sure at this moment.  I can see arguments for either way.